mantidproject/mantid
# ILL TOF SANS wavelength range

**Description of work.**
Wavelength range is exposed to the ILL SANS algorithm and the interface.
It performs a cut using the new C++ version of CropWorkspaceRagged before integration.



**To test:**
Use the run in unit test directory, under ILL/D33.
```
SANSILLReduction(Run='093410', ProcessAs='Sample', OutputWorkspace='sample')
MaskBTP(Workspace='sample', Components='back_detector', Tube='110-140', Pixel='55-73') 
MaskBTP(Workspace='sample', Components='back_detector', Pixel='127') 
SANSILLIntegration(InputWorkspace='sample', OutputWorkspace='iq', WavelengthRange=[1,6])
```
Make sure the iq workspace x-axis spans from around 0.01 to 1 A^-1.


Part of  #26906. 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
